Scripts/Utilities/AssetDropUtils.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+        internal static AudioClip AttachAudioClip(GameObject go, AssetData data)
+        {
+            var source = go.GetComponent<AudioSource>();
+            if (source == null)
+                source = Undo.AddComponent<AudioSource>(go);
+            
+            source.clip = (AudioClip)data.asset;
+
+            Undo.IncrementCurrentGroup();
+            return source.clip;
+        }
+
+        internal static VideoClip AttachVideoClip(GameObject go, AssetData data)
+        {
+            var player = go.GetComponent<VideoPlayer>();
+            if (player == null)
+                player = Undo.AddComponent<VideoPlayer>(go);
+
+            Undo.RecordObject(player, k_AssignVideoClipUndoLabel);
+            player.clip = (VideoClip)data.asset;
+
+            Undo.IncrementCurrentGroup();
+            return player.clip;
+        }
+
+        internal static Type AttachScript(GameObject go, AssetData data)
+        {
+            var script = (MonoScript)data.asset;
+            var type = script.GetClass();
+            Undo.AddComponent(go, type);
+            return type;
+        }
+
+        internal static Material AssignMaterial(GameObject go, AssetData data)
+        {
+            var renderer = go.GetComponent<Renderer>();
+
+            if (renderer != null)
+            {
+                Undo.RecordObject(go, k_AssignMaterialUndoLabel);
+                renderer.sharedMaterial = (Material)data.asset;
+                Undo.IncrementCurrentGroup();
+                return renderer.sharedMaterial;
+            }
+
+            return null;
+        }
+
+        internal static Shader AssignMaterialShader(GameObject go, AssetData data)
+        {
+            var renderer = go.GetComponent<Renderer>();
+
+            if (renderer != null)
+            {
+                Undo.RecordObject(go, k_AssignMaterialUndoLabel);
+                // this warns that we're leaking materials into the scene,
+                // and creates a new instance, but we don't want to change
+                // the shader on the shared material here.
+                var shader = (Shader)data.asset;
+                renderer.material.shader = shader;
+                Undo.IncrementCurrentGroup();
+                return shader;
+            }
+
+            return null;
+        }
+
+        internal static PhysicMaterial AssignColliderPhysicMaterial(GameObject go, AssetData data)
+        {
+            var collider = go.GetComponent<Collider>();
+
+            if(collider != null)
+            {
+                Undo.RecordObject(go, k_AssignPhysicMaterialUndoLabel);
+                collider.material = (PhysicMaterial)data.asset;
+                Undo.IncrementCurrentGroup();
+
+                return collider.material;
+            }
+
+            return null;
+        }
+
+        internal static Font AssignFontOnChildren(GameObject go, AssetData data)
+        {
+            var text = go.GetComponentInChildren<TextMesh>();
```

Comment:
What if there is more than one child?

Reply:
i originally had it that way, and for some odd reason, setting a new font on an object that only had one textmesh produced garbled text somehow ?  once i just changed back to a single component, it went away.

I'll revisit that.  that bug seems unlikely and maybe unrelated.